src_c/base.c

```diff
@@ -2097,7 +2097,7 @@ MODINIT_DEFINE(base)
 {
     PyObject *module, *apiobj, *atexit;
     PyObject *atexit_register;
-#if !defined(BUILD_STATIC) || defined(NO_PYGAME_C_API)
+#if !(defined(BUILD_STATIC) && defined(NO_PYGAME_C_API))
```

Comment:
I understand the check for `BUILD_STATIC`, but I'm not sure why `NO_PYGAME_C_API` is relevant here

Reply:
i think when pygame c api is active the pgExc_SDLError becomes a macro that point to slot 0 of base.c, which may not work with dynamic linking and wasm ( need a global, you cannot address a pointer from another C module without using a js trampoline ( the one that handle dlopen ) to bridge the wasm memories ).

Reply:
What about the case where `BUILD_STATIC` is defined but `NO_PYGAME_C_API` is not?

Reply:
that would be possible (far) future dynamic linking ( converting the whole libpygame.a into a pygame.so module ).